### ISSUE TITLE
Git operations over ssh gives "Gitea: Internal error" and "Failed to get repository: dial tcp 127.0.0.1:3306: connect: connection refused"

### ISSUE THREAD

**Issue by hak8or** [go-gitea/gitea#4092]
- Gitea version (or commit ref): 1.4.1
- Git version: 2.17.1
- Operating system: Arch
- Database (use `[x]`):
  - [ ] PostgreSQL
  - [ ] MySQL
  - [ ] MSSQL
  - [x] SQLite
- Can you reproduce the bug at https://try.gitea.io:
  - [ ] Yes (provide example URL)
  - [ ] No
  - [x] Not relevant
- Log gist:

## Description
After updating (I didn't keep track of the previous version) I am unable to do any git pull/push/fetch on any of my repositories over ssh. I am able to do git operations over http for some reason though. The web GUI works fine, and I am able to access all my repositories and create repositories through the web GUI as well. 

I am also puzzled as to why the error mentions connecting even though I am using sqlite3, not postgres/mysql/etc.

Only git operations over SSH don't seem to work.

For example:
```bash
λ git clone gitea@gitea.hak8or.com:hak8or/foo.git
Cloning into 'foo'...
Gitea: Internal error
Failed to get repository: dial tcp 127.0.0.1:3306: connect: connection refused
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.
```

```bash
λ git clone http://gitea.hak8or.com/hak8or/foo.git
Cloning into 'foo'...
remote: Counting objects: 5, done.
remote: Compressing objects: 100% (4/4), done.
remote: Total 5 (delta 0), reused 0 (delta 0)
Unpacking objects: 100% (5/5), done.
```

I can access gitea via ssh though:
```bash
λ ssh gitea@gitea.hak8or.com
PTY allocation request failed on channel 0
Hi there, You've successfully authenticated, but Gitea does not provide shell access.
If this is unexpected, please log in with password and setup Gitea under another user.
Connection to gitea.hak8or.com closed.
```

## app.ini
```
APP_NAME = Hak8or Home Repos
RUN_USER = gitea
RUN_MODE = prod

[repository]
ROOT = /var/lib/gitea/repos

[server]
DOMAIN           = gitea.hak8or.com
PROTOCOL         = http
ROOT_URL         = %(PROTOCOL)s://%(DOMAIN)s
HTTP_PORT        = 3900
SSH_DOMAIN       = gitea.hak8or.com
STATIC_ROOT_PATH = /var/lib/gitea/

[database]
DB_TYPE             = sqlite3
PATH                = /var/lib/gitea/data/gitea.db

[security]
INSTALL_LOCK   = true
SECRET_KEY     = ---removed---
INTERNAL_TOKEN = ---removed---

[log]
ROOT_PATH = /var/log/gitea/
LEVEL     = Trace
```

## Logs
I do not see any log updates via ```journalctl -u gitea``` but sshd does see the attempt.

```bash
[root@Gitea ~]# journalctl -u sshd --since "-1 minute" -f
-- Logs begin at Wed 2018-05-30 19:53:01 UTC. --
Jun 01 02:33:27 Gitea sshd[14227]: Accepted publickey for gitea from 192.168.1.1 port 7183 ssh2: RSA SHA256:uNQY0bMlMR5K8OULv9HtQZwCou8NUd0ULoVIW6mjQiI
Jun 01 02:33:27 Gitea sshd[14227]: pam_unix(sshd:session): session opened for user gitea by (uid=0)
Jun 01 02:33:28 Gitea sshd[14227]: pam_unix(sshd:session): session closed for user gitea
````

**Comment by lafriks** [comment]
Seems to be configuration issue as `dial tcp 127.0.0.1:3306` means that gitea tries to connect to MySQL and can not...

**Comment by techknowlogick** [comment]
Can you go into your admin dashboard and run `Update the '.ssh/authorized_keys' file with Gitea SSH keys.`. I have a suspicion that because you recently upgraded perhaps the old configuration may be linked somewhere.

**Comment by lafriks** [comment]
Check authorized_keys file contents to see if it point correct gitea binary and correct app.ini location

**Comment by techknowlogick** [comment]
Also check .ssh/environment file (if exists) as it might set `GITEA_CUSTOM` environment variable

**Comment by adrinux** [comment]
Also seeing this issue using system ssh server (built in disabled). Can clone via http but not ssh. ssh logs indicate gitea user authenticates successfully but git reports:
```
fatal: Could not read from remote repository.

Please make sure you have the correct access rights
and the repository exists.

```
As for others. Cycling the authorized_keys file in the admin section makes no difference (backup is identical). authorized_keys points to correct binary and app.ini. The authorized_key listed is the one I uploaded to my user account.

Here's debug output from ssh -T -vvv gitea@my.gitea.server  
```
debug1: Authentication succeeded (publickey).
Authenticated to [REDACTED]
debug1: channel 0: new [client-session]
debug3: ssh_session2_open: channel_new: 0
debug2: channel 0: send open
debug3: send packet: type 90
debug1: Requesting no-more-sessions@openssh.com
debug3: send packet: type 80
debug1: Entering interactive session.
debug1: pledge: network
debug3: receive packet: type 80
debug1: client_input_global_request: rtype hostkeys-00@openssh.com want_reply 0
debug3: receive packet: type 4
debug1: Remote: Forced command.
debug3: receive packet: type 4
debug1: Remote: Port forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: X11 forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: Agent forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: PTY allocation disabled.
debug3: receive packet: type 4
debug1: Remote: Forced command.
debug3: receive packet: type 4
debug1: Remote: Port forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: X11 forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: Agent forwarding disabled.
debug3: receive packet: type 4
debug1: Remote: PTY allocation disabled.
debug3: receive packet: type 91
debug2: channel_input_open_confirmation: channel 0: callback start
debug2: fd 3 setting TCP_NODELAY
debug3: ssh_packet_set_tos: set IP_TOS 0x08
debug2: client_session2_setup: id 0
debug1: Sending environment.
debug3: Ignored env XDG_CURRENT_DESKTOP
debug3: Ignored env DBUS_STARTER_ADDRESS
debug1: Sending env LANG = en_GB.UTF-8
debug2: channel 0: request env confirm 0
debug3: send packet: type 98
debug3: Ignored env DISPLAY
debug3: Ignored env PWD
debug3: Ignored env LOGNAME
debug3: Ignored env SHLVL
debug3: Ignored env INVOCATION_ID
debug3: Ignored env MANDATORY_PATH
debug3: Ignored env XDG_VTNR
debug3: Ignored env XAUTHORITY
debug3: Ignored env JOURNAL_STREAM
debug3: Ignored env GTK_IM_MODULE
debug3: Ignored env COLORTERM
debug3: Ignored env XDG_SESSION_ID
debug3: Ignored env DESKTOP_SESSION
debug3: Ignored env IM_CONFIG_PHASE
debug3: Ignored env XDG_SESSION_DESKTOP
debug3: Ignored env GDMSESSION
debug3: Ignored env TEXTDOMAINDIR
debug3: Ignored env USERNAME
debug3: Ignored env WINDOWPATH
debug3: Ignored env TEXTDOMAIN
debug3: Ignored env GNOME_DESKTOP_SESSION_ID
debug3: Ignored env DBUS_SESSION_BUS_ADDRESS
debug3: Ignored env VTE_VERSION
debug3: Ignored env DBUS_STARTER_BUS_TYPE
debug3: Ignored env XDG_DATA_DIRS
debug3: Ignored env _
debug3: Ignored env QT4_IM_MODULE
debug3: Ignored env CLUTTER_IM_MODULE
debug3: Ignored env XDG_MENU_PREFIX
debug3: Ignored env QT_ACCESSIBILITY
debug3: Ignored env SHELL
debug3: Ignored env QT_IM_MODULE
debug3: Ignored env XDG_SESSION_TYPE
debug3: Ignored env XDG_CONFIG_DIRS
debug3: Ignored env MANAGERPID
debug3: Ignored env SSH_AGENT_PID
debug3: Ignored env GTK_MODULES
debug3: Ignored env SESSION_MANAGER
debug3: Ignored env SSH_AUTH_SOCK
debug3: Ignored env PATH
debug3: Ignored env LIBVIRT_DEFAULT_URI
debug3: Ignored env TERM
debug3: Ignored env HOME
debug3: Ignored env USER
debug3: Ignored env XMODIFIERS
debug3: Ignored env XDG_RUNTIME_DIR
debug3: Ignored env DEFAULTS_PATH
debug3: Ignored env GPG_AGENT_INFO
debug3: Ignored env XDG_SEAT
debug3: Ignored env TILIX_ID
debug3: Ignored env OLDPWD
debug3: Ignored env EDITOR
debug3: Ignored env VISUAL
debug3: Ignored env PAGER
debug3: Ignored env LESS
debug3: Ignored env LESSOPEN
debug3: Ignored env TMPDIR
debug3: Ignored env PYTHONPATH
debug3: Ignored env ANSIBLE_LIBRARY
debug3: Ignored env MANPATH
debug3: Ignored env GOPATH
debug3: Ignored env GOBIN
debug3: Ignored env LESS_TERMCAP_mb
debug3: Ignored env LESS_TERMCAP_md
debug3: Ignored env LESS_TERMCAP_me
debug3: Ignored env LESS_TERMCAP_se
debug3: Ignored env LESS_TERMCAP_so
debug3: Ignored env LESS_TERMCAP_ue
debug3: Ignored env LESS_TERMCAP_us
debug3: Ignored env LS_COLORS
debug3: Ignored env GREP_COLOR
debug3: Ignored env GREP_COLORS
debug3: Ignored env NVM_CD_FLAGS
debug3: Ignored env NVM_DIR
debug3: Ignored env NVM_BIN
debug2: channel 0: request shell confirm 1
debug3: send packet: type 98
debug2: channel_input_open_confirmation: channel 0: callback done
debug2: channel 0: open confirm rwindow 0 rmax 32768
debug2: channel 0: rcvd adjust 2097152
debug3: receive packet: type 99
debug2: channel_input_status_confirm: type 99 id 0
debug2: shell request accepted on channel 0
debug3: receive packet: type 96
debug2: channel 0: rcvd eof
debug2: channel 0: output open -> drain
debug2: channel 0: obuf empty
debug2: channel 0: close_write
debug2: channel 0: output drain -> closed
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype exit-status reply 0
debug3: receive packet: type 98
debug1: client_input_channel_req: channel 0 rtype eow@openssh.com reply 0
debug2: channel 0: rcvd eow
debug2: channel 0: close_read
debug2: channel 0: input open -> closed
debug3: receive packet: type 97
debug2: channel 0: rcvd close
debug3: channel 0: will not send data after close
debug2: channel 0: almost dead
debug2: channel 0: gc: notify user
debug2: channel 0: gc: user detached
debug2: channel 0: send close
debug3: send packet: type 97
debug2: channel 0: is dead
debug2: channel 0: garbage collecting
debug1: channel 0: free: client-session, nchannels 1
debug3: channel 0: status: The following connections are open:
  #0 client-session (t4 r0 i3/0 o3/0 fd -1/-1 cc -1)

debug3: send packet: type 1
Transferred: sent 2720, received 3004 bytes, in 1.0 seconds
Bytes per second: sent 2766.3, received 3055.1
debug1: Exit status 1
```
Hope this helps.

**Comment by hak8or** [comment]
Apologies for not getting back to @lafriks and @techknowlogick for so long. Thank you for the suggestions.

I checked the ```authorized_keys``` file and it looked ok but I can't say with certainty. I did not see any ```.ssh/environment``` file. In the end I just deleted the ```authorized_keys``` file, re-imported my ssh key via gitea, and it all seems to work now. I am still not clear why it wasn't worked earlier, but I assume that an update went wrong and the command in authorized_keys got de-sync'd with what it should have been.

Closing since this issue has been resolved.
